browse/factory.py

```diff
@@ -1,79 +1,54 @@
 """Application factory for browse service components."""
 from functools import partial
-from typing import Any
-from flask import Flask, url_for
-from browse.domain.identifier import canonical_url
-from browse.util.clickthrough import create_ct_url
-from browse.util.id_patterns import do_dois_id_urls_to_tags, do_id_to_tags, \
-    do_dois_arxiv_ids_to_tags
+from flask import Flask
+from arxiv.base.urls import canonical_url, clickthrough_url, urlizer
 from browse.routes import ui
 from browse.services.database import models
 from browse.services.util.email import generate_show_email_hash
-from browse.filters import line_feed_to_br, tex_to_utf, entity_to_utf, \
-    single_doi_url
+from browse.filters import entity_to_utf
+from browse.services.listing.fake_listings import FakeListingFilesService
 
 from arxiv.base.config import BASE_SERVER
 from arxiv.base import Base
+from arxiv.users.auth import Auth
 
 
 def create_web_app() -> Flask:
     """Initialize an instance of the browse web application."""
     app = Flask('browse', static_folder='static', template_folder='templates')
-    app.config.from_pyfile('config.py')
+    app.config.from_pyfile('config.py')  # type: ignore
 
     # TODO Only needed until this route is added to arxiv-base
     if 'URLS' not in app.config:
         app.config['URLS'] = []
     app.config['URLS'].append(
         ('search_archive', '/search/<archive>', BASE_SERVER))
 
-    models.init_app(app)
+    models.init_app(app)  # type: ignore
 
     Base(app)
+    Auth(app)
     app.register_blueprint(ui.blueprint)
 
-    ct_url_for = partial(create_ct_url, app.config.get(
-        'CLICKTHROUGH_SECRET'), url_for)
+    app.config['listing_service'] = FakeListingFilesService()
```

Comment:
This is not a pattern that we want to use. I realize that this is a fake service, but let's not let this stick around. The ``config`` is for configuration parameters. Instances of services can be bound to the appropriate context using the [``g`` object](http://flask.pocoo.org/docs/1.0/appcontext/#storing-data).